.github/workflows/benchmark-cachegrind.yml

```diff
@@ -0,0 +1,102 @@
+name: Luau Benchmarks (CacheGrind)
+
+on:
+  pull_request:
+    paths-ignore:
+      - "docs/**"
+      - "papers/**"
+      - "rfcs/**"
+      - "*.md"
+      - "prototyping/**"
+
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - "docs/**"
+      - "papers/**"
+      - "rfcs/**"
+      - "*.md"
+      - "prototyping/**"
+
+jobs:
+  benchmarks-run:
+    name: Run ${{ matrix.bench.title }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        engine:
+          - { channel: stable, version: latest }
```

Comment:
Do we need this matrix variable? Might be a leftover from internal setup.

Reply:
We will likely be running  the benchmarks on different OSes, in which case we'd need the matrix. Still investigating as to whether we would run the benchmarks in a single workflow or in separate workflows (if there are sufficient differences in the workflow steps for the different OSes)

Reply:
Guessing you mean `engine`, actually, nope! Removed it